test/parallel/test-dotenv-edge-cases.js

```diff
@@ -200,4 +200,35 @@ describe('.env supports edge cases', () => {
     assert.strictEqual(child.code, 9);
     assert.match(child.stderr, /bad option: --env-file-ABCD/);
   });
+
+  it('should handle invalid syntax in .env file', async () => {
+    const code = `
+      const resutl = util.parseEnv(
+        \`
+        foo
+
+        bar
+        baz=whatever
+        VALID_AFTER_INVALID=test
+        multiple_invalid
+        lines_without_equals
+        ANOTHER_VALID=value
+        \`
+      );
```

Comment:
nit: we can use an array to avoid passing the extra white space, not sure if that matters
```suggestion
      const resutl = util.parseEnv([
        'foo',
        '',
        'bar',
        'baz=whatever',
        'VALID_AFTER_INVALID=test',
        'multiple_invalid',
        'lines_without_equals',
        'ANOTHER_VALID=value',
      ].join('\n'))
```